src/dotnet/ToolPackage/PackageLocation.cs

```diff
@@ -8,16 +10,33 @@ internal class PackageLocation
     {
         public PackageLocation(
             FilePath? nugetConfig = null,
-            DirectoryPath? rootConfigDirectory = null, 
-            string[] additionalFeeds = null)
+            DirectoryPath? rootConfigDirectory = null,
+            string[] additionalFeeds = null,
+            bool disableParallel = false,
+            bool NoCache = false,
+            bool ignoreFailedSources = false)
         {
             NugetConfig = nugetConfig;
             RootConfigDirectory = rootConfigDirectory;
             AdditionalFeeds = additionalFeeds ?? Array.Empty<string>();
+            DisableParallel = disableParallel;
+            IgnoreFailedSources = IgnoreFailedSources;
+        }
+
+        public PackageLocation(AppliedOption appliedOption)
```

Comment:
I'm not sure I like the CLI options "leaking" into the tool package types.  This is now another place we need to keep in sync with the command line parser for the commands.

Reply:
actually that's should the right thing. Just put it is in CreateToolPackageStoreAndInstaller Factory